nixos/modules/services/development/fossil-server.nix

```diff
@@ -0,0 +1,146 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+  cfg = config.services.fossil;
+in
+{
+
+  ###### interface
+
+  options.services.fossil = {
+
+    enable = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+          Enable the Fossil server, which will start an HTTP server to
+          publish Fossil repositories.
+        '';
+    };
+
+    repository = mkOption {
+      type = types.str;
+      default = "/srv/fossil";
+      example = "/fossils/repo.fossil";
+      description = ''
+          The REPOSITORY can be a directory that contains one or more
+          repositories with names ending in ".fossil" or a single file.
+        '';
+    };
+
+    baseurl = mkOption {
+      type = types.str;
+      default = "";
+      example = "example.com";
+      description = "Use URL as the base (useful for reverse proxies)";
+    };
+
+    extroot = mkOption {
+      type = types.str;
+      default = "";
+      description = "Document root for the /ext extension mechanism";
+    };
+
+    files = mkOption {
+      type = types.str;
+      default = "";
+      description = "Comma-separated list of glob patterns for static files";
+    };
+
+    https = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+          Indicates that the input is coming through a reverse proxy
+          that has already translated HTTPS into HTTP.
+        '';
+    };
+
+    jsmode = mkOption {
+      type = types.str;
+      default = "";
+      example = "inline";
+      description = ''
+          Determine how JavaScript is delivered with pages.
+          Mode can be one of:
+            inline       All JavaScript is inserted inline at
+                         the end of the HTML file.
+            separate     Separate HTTP requests are made for
+                         each JavaScript file.
+            bundled      One single separate HTTP fetches all
+                         JavaScript concatenated together.
+        '';
+    };
+
+    maxLatency = mkOption {
+      type = types.int;
+      default = 0;
+      example = 60;
+      description = "Do not let any single HTTP request run for more than N seconds";
+    };
+
+    nocompress = mkOption {
+      type = types.bool;
+      default = false;
+      description = "Do not compress HTTP replies";
+    };
+
+    nojail = mkOption {
```

Comment:
From *quickly* reviewing documentation it seems like this option doesn't provide any real value when you're using `systemd`.

Reply:
I just removed it, never used it before so would not know how to test it. If someone needs it in the future then it could be investigated.